spec/models/concerns/allowance_checker_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe AllowanceChecker do
+  subject(:student) { create(:student) }
+
+  let(:mef) { create(:mef, code: "123") }
+  let(:classe) { create(:classe, mef: mef) }
+  let(:schooling) { create(:schooling, student: student, classe: classe) }
+
+  before do
+    mef.wage.update!(yearly_cap: 100, daily_rate: 10) # dodgy
+  end
```

Comment:
Pourquoi ne pas l'initialisé directement via la factory en passant le modèle `wage` ?

Reply:
parce que le lien entre `wage` et `mef` n'est pas un one-to-one donc c'est plus facile comme ça (https://github.com/betagouv/aplypro/blob/payments/allowance-rework/spec/factories/mefs.rb#L11-L13)

Reply:
J'encapsulerai tout de même cette complexité dans la factory, c'est fait pour ça. (c'est clairement pas critique, juste les setups de modèles dans les tests peuvent devenir lourds, faire l'effort de faire des bonnes factories simplifie grandement, et comme ça touche à la couche modèle je trouve que ça mérite de faire l'effort).

Reply:
je regarde